File: back-end/model/ChamCong.py
from sqlalchemy import Column, String, Date, ForeignKey, NVARCHAR
from sqlalchemy.dialects.mssql import TIME
from sqlalchemy.orm import declarative_base, relationship
from database import Base

class ChamCong(Base):
    __tablename__= 'ChamCong'
    MaChamCong = Column(String(30), primary_key=True)
    MaNV = Column(String(20), ForeignKey('NhanVien.MaNV'), nullable=True)
    NhanVien = relationship("NhanVien")
    MaQL = Column(String(20), ForeignKey('QuanLy.MaQL'), nullable=True)
    QuanLy = relationship("QuanLy")
    MaCa = Column(String(20), ForeignKey('CaLam.MaCa'), nullable=True)
    CaLam = relationship("CaLam")
    NgayChamCong = Column(Date, nullable=False)
    GioVao = Column(TIME(precision=0), nullable=True)
    GioRa = Column(TIME(precision=0), nullable=True)
    TrangThai = Column(NVARCHAR(50), nullable=False)
    DiaDiemChamCong = Column(NVARCHAR(255), nullable=True) 

File: front-end/lib/models/chamcong.dart
class ChamCong {
  final String id;          // MaChamCong
  final String userId;      // MaNV
  final String? maQL;       // MaQL
  final String maCa;       // MaCa
  final DateTime ngayChamCong;
  final String gioVao;
  final String gioRa;
  final String trangThai;
  final String diaDiemChamCong;

  ChamCong({
    required this.id,
    required this.userId,
    this.maQL,
    required this.maCa,
    required this.ngayChamCong,
    required this.gioVao,
    required this.gioRa,
    required this.trangThai,
    required this.diaDiemChamCong,
  });

  factory ChamCong.fromJson(Map<String, dynamic> json) {
    return ChamCong(
      id: json['id'] ?? '',
      userId: json['userId'] ?? '',
      maQL: json['maQL'],
      maCa: json['maCa']?.toString() ?? '',
      ngayChamCong: DateTime.parse(json['ngayChamCong']),
      gioVao: json['gioVao'],
      gioRa: json['gioRa'],
      trangThai: json['trangThai'] ?? '',
      diaDiemChamCong: json['diaDiemChamCong'] ?? '',
    );
  }
}
